pkg/util/mount/mount_linux.go

```diff
@@ -430,3 +430,40 @@ func (mounter *SafeFormatAndMount) getDiskFormat(disk string) (string, error) {
 	// and MD RAID are reported as FSTYPE and caught above).
 	return "unknown data, probably partitions", nil
 }
+
+// IsSamefsGroup is called only for requests to mount an already mounted
+// volume. It checks if fsGroup of new mount request is the same or not.
+// It return false if it not the same.
+func IsSamefsGroup(dir string, fsGroup *int64) (bool, error) {
+	s := &syscall.Stat_t{}
+	if err := syscall.Stat(dir, s); err != nil {
+		return false, err
+	}
+	gidNew := 0
+	if fsGroup != nil {
+		gidNew = int(*fsGroup)
```

Comment:
fsGroup == nil means it is not set and kubelet would use default which is 0
*fsGroup == 0 means default GID which belong to root

So there might be different initial cases. 

Please follow this example:
1.1. pod 1 mounts local volume with fsgroup (not set) or set to 0, mount successful
1.2. pod 2 mount  local volume with fsgroup not set or set to 0, mount successful as when not set gid == 0

2.1 pod 1 mounts local volume with fsgroup == 2000 mount sucessful because we do not have any entries in mount table, so we do not check for matching fsgroup
2.2 pod2 mount local volume with fsgroup == 2000 mount sucessful because even though we detected that the volume has laready been mounted, there is match of fsgroups
2.3 pod3 mount local volume with fsgroup == 2000 mount fails because we detected that the volume has already been mounted and fsgroups do not match.

Now here is the corner case which I was trying to address by another PR.
pod 1 and pod 2 were terminated, but the volume mount point still have old GID of 2000.
New pod11 starts and mounts the local volume successfully because no mount entries, even though it uses GID let's say 3000, and we do not check for matching fsgroup, but pod22 mount fails because its GID of 3000 does not match to old GID of 2000.

It looks like we either need to re-write GID on sucessful initial mount even though it is not specified or reset GID on last unmount of the volume